src/YearnFed.sol

```diff
@@ -0,0 +1,162 @@
+        require(msg.sender == gov, "ONLY GOV");
+        require(newMaxLossBpContraction <= 10000);
+        maxLossBpContraction = newMaxLossBpContraction;
+    }
+
+    /**
+    @notice Method for governance to set max loss in basis points, when taking profit from yearn vault
+    @param newMaxLossBpTakeProfit new maximally allowed loss in Bp 1 = 0.001%
+    */
+    function setMaxLossBpTakeProfit(uint newMaxLossBpTakeProfit) public {
+        require(msg.sender == gov, "ONLY GOV");
+        require(newMaxLossBpTakeProfit <= 100000);
+        maxLossBpTakeProfit = newMaxLossBpTakeProfit;
+    }
+
+    /**
+    @notice Method for withdrawing any token from the contract to governance. Should only be used in emergencies.
+    @param token Address of token contract to withdraw to gov
+    @param amount Amount of tokens to withdraw
+    */
+    function emergencyWithdraw(address token, uint amount) public{
+        require(msg.sender == gov, "ONLY GOV");
+        require(token != address(vault));
+        IERC20(token).transfer(gov, amount);
+    }
+
+    /**
+    @notice Method for current chair of the Yearn FED to resign
+    */
+    function resign() public {
+        require(msg.sender == chair, "ONLY CHAIR");
+        chair = address(0);
+    }
+
+    /**
+    @notice Deposits amount of underlying tokens into yEarn vault
+
+    @param amount Amount of underlying token to deposit into yEarn vault
+    */
+    function expansion(uint amount) public {
+        require(msg.sender == chair, "ONLY CHAIR");
+        //Alternatively set amount to max uint if over deposit limit,
+        //as that supplies greatest possible amount into vault
+        /*
+        if( amount > _maxDeposit()){
+            amount = type(uint256).max;
+        }
+        */
+        require(amount <= _maxDeposit(), "AMOUNT TOO BIG"); // can't deploy more than max
+        underlying.mint(address(this), amount);
+        uint shares = vault.deposit(amount, address(this));
+        require(shares == 0, 'Supplying failed'); //Probably an unnecessary require
+        supply = supply + amount;
+        emit Expansion(amount);
+    }
+
+    /**
+    @notice Withdraws an amount of underlying token to be burnt, contracting DOLA supply
+    
+    @dev Its recommended to always broadcast withdrawl transactions(contraction & takeProfits)
+    through a frontrun protected RPC like Flashbots RPC.
+    
+    @param amount The amount of underlying tokens to withdraw. Note that more tokens may
+    be withdrawn than requested, as price is calculated by debts to strategies, but strategies
+    may have outperformed price of underlying token.
+    */
+    function contraction(uint amount) public {
+        require(msg.sender == chair, "ONLY CHAIR");
+        uint underlyingWithdrawn = _withdrawAmountUnderlying(amount, maxLossBpContraction);
+        require(underlyingWithdrawn <= supply, "AMOUNT TOO BIG"); // can't burn profits
```

Comment:
could avoid forcing a revert here by doing:
```
uint toBurn = Math.min(supply, underlyingWithdrawn)
```

Reply:
Thank you! Went with an if statement, as we want to transfer any surplus to governance.

Reply:
The remaining supply would be left as bad debt on the balance sheet, but it may not be a bad idea to have a governance function for bookkeeping sake.